lib/kwtool_web/controllers/session_controller.ex

```diff
@@ -0,0 +1,46 @@
+defmodule KwtoolWeb.SessionController do
+  use KwtoolWeb, :controller
+
+  alias Kwtool.Accounts
+  alias Kwtool.Accounts.Guardian
+  alias Kwtool.Accounts.Schemas.User
+
+  plug :put_layout, "auth.html"
+
+  def new(conn, _) do
+    changeset = Accounts.change_user(%User{})
+    maybe_user = Guardian.Plug.current_resource(conn)
+
+    if maybe_user do
+      redirect(conn, to: "/home")
+    else
+      render(conn, "new.html", changeset: changeset, action: Routes.session_path(conn, :create))
+    end
+  end
+
+  def create(conn, %{"user" => %{"email" => email, "password" => plain_text_password}}) do
+    email
+    |> Accounts.authenticate_user(plain_text_password)
+    |> login_reply(conn)
```

Comment:
I think this might get harder to follow if we have more functions later. we can just do

```elixir
case Accounts.authenticate_user(email, plain_text_password) do
  {:ok, user} ->
    ...
      
  {:error, _reason} -> 
    ...
end
```

Reply:
Updated